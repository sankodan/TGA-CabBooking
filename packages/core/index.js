import { theme } from './src/theme'
import { firebaseConfig } from './src/config/firebase'
import {
  signUpUser,
  logoutUser,
  loginUser,
  sendEmailWithPassword
} from './src/api/auth'
import {
  passwordValidator,
  nameValidator,
  emailValidator
} from './src/helpers'

module.exports = {
  theme,
  firebaseConfig,
  signUpUser,
  logoutUser,
  loginUser,
  sendEmailWithPassword,
  passwordValidator,
  nameValidator,
  emailValidator
}
