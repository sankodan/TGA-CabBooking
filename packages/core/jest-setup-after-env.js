import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import '@testing-library/jest-native/extend-expect'
import { NativeModules } from 'react-native';

configure({ adapter: new Adapter() })

jest.mock('react-native-blob-util', () => {
  return {
    DocumentDir: () => {},
    polyfill: () => {},
    dirs:()=>{}
  }
});

jest.mock('react-native/Libraries/EventEmitter/NativeEventEmitter')
jest.mock('@react-native-firebase/auth', () => {
  return () => ({
    createUserAndRetrieveDataWithEmailAndPassword: jest.fn(() => Promise.resolve(true)),
    sendPasswordResetEmail: jest.fn((email) => {
      if (email)  Promise.resolve(true)
      else  throw new Error()
    }),
    createUserWithEmailAndPassword: jest.fn((email, password) => {
      if (email && password) Promise.resolve(true)
      else throw new Error()
    }),
    fetchSignInMethodsForEmail: jest.fn(() => Promise.resolve(true)),
    signOut: jest.fn(() => {
      Promise.resolve(false);
    }),
    onAuthStateChanged: jest.fn(),
    signInWithEmailAndPassword: jest.fn((email, password) => {
      if (email && password) {
       Promise.resolve(true)
      } else throw new Error()
    }),
    currentUser: {
      sendEmailVerification: jest.fn(() => Promise.resolve(true)),
      updateProfile:jest.fn()
    },
  })
})

jest.mock('redux-persist-filesystem-storage', () => {
  return {
    dirs:()=>{}
  }
});

jest.mock('redux-persist', () => {
  return {
    getItem:()=>{},
    persistReducer:()=> {
      return ()=>{}
    },
    persistStore:()=> {
      return ()=>{}
    }
  }
});

jest.mock('redux-persist/integration/react', () => ({
  PersistGate: props => props.children,
}));


NativeModules.RNCAsyncStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  mergeItem: jest.fn(),
  clear: jest.fn(),
  getAllKeys: jest.fn(),
  flushGetRequests: jest.fn(),
  multiGet: jest.fn(),
  multiSet: jest.fn(),
  multiRemove: jest.fn(),
  multiMerge: jest.fn(),
};

NativeModules.RNEncryptedStorage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  mergeItem: jest.fn(),
  clear: jest.fn(),
  getAllKeys: jest.fn(),
  flushGetRequests: jest.fn(),
  multiGet: jest.fn(),
  multiSet: jest.fn(),
  multiRemove: jest.fn(),
  multiMerge: jest.fn(),
};
