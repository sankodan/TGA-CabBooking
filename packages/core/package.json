{
  "name": "@sankodan/com.cabbooking.core",
  "version": "1.0.3",
  "description": "",
  "main": "index.js",
  "scripts": {
    "lint": "eslint src --fix",
    "test": "jest src -u --coverage"
  },
  "author": "",
  "license": "ISC",
  "repository": {
    "type": "git",
    "url": "https://github.com/sankodan/TGA-CabBooking.git"
  },
  "publishConfig": {
    "@sankodan:registry": "https://npm.pkg.github.com"
  },
  "dependencies": {
    "@react-native-firebase/app": "^15.2.0",
    "@react-native-firebase/auth": "^15.2.0",
    "@reduxjs/toolkit": "1.8.4",
    "axios": "0.27.2",
    "react-native-encrypted-storage": "4.0.2",
    "react-native-paper": "^4.12.4",
    "react-redux": "^7.1.1",
    "redux": "^4.0.4",
    "redux-persist": "^6.0.0",
    "redux-persist-filesystem-storage": "4.0.0",
    "redux-promise-middleware": "^6.1.1",
    "redux-thunk": "^2.3.0"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@babel/runtime": "^7.12.5",
    "@react-native-community/eslint-config": "^2.0.0",
    "@testing-library/jest-native": "4.0.5",
    "@testing-library/react-native": "10.1.0",
    "axios-mock-adapter": "1.21.2",
    "babel-jest": "^26.6.0",
    "detox": "19.7.1",
    "enzyme": "^3.11.0",
    "enzyme-adapter-react-16": "^1.15.6",
    "eslint": "^7.11.0",
    "eslint-plugin-detox": "^1.0.0",
    "jest": "26.6.0",
    "metro-react-native-babel-preset": "^0.70.3",
    "react-dom": "^18.2.0",
    "react-native-monorepo-tools": "^1.1.3",
    "react-test-renderer": "18.0.0",
    "redux-mock-store": "1.5.4"
  }
}
