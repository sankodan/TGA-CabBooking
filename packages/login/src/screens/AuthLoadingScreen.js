import React from 'react'
import { ActivityIndicator } from 'react-native'
import auth from '@react-native-firebase/auth'
import { Background } from '@sankodan/com.cabbooking.components'
import { theme } from '@sankodan/com.cabbooking.core'

export default function AuthLoadingScreen ({ navigation }) {
  const onAuthStateChanged = (user) => {
    if (user) {
      // User is logged in
      navigation.reset({
        index: 0,
        routes: [{ name: 'Dashboard' }]
      })
    } else {
      // User is not logged in
      navigation.reset({
        index: 0,
        routes: [{ name: 'StartScreen' }]
      })
    }
  }

  React.useEffect(() => {
    const subscriber = auth().onAuthStateChanged(onAuthStateChanged)
    return subscriber // unsubscribe on unmount
  }, [])

  return (
    <Background>
      <ActivityIndicator size="large" color={theme.colors.primary} />
    </Background>
  )
}
