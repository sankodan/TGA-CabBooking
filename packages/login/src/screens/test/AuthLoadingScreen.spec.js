import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react-native'
import AuthLoadingScreen from '../AuthLoadingScreen'
import { shallow } from 'enzyme'
import { logoutUser } from '@sankodan/com.cabbooking.core'

describe('AuthLoadingScreen', () => {
  let toJSONObject
  describe('AuthLoadingScreen', () => {
    const navigate = jest.fn()
    const popToTop = jest.fn()
    const push = jest.fn()
    const navigation = {
      navigate,
      reset: jest.fn(),
      popToTop,
      push
    }
    beforeEach(() => {
      const { toJSON } = render(<AuthLoadingScreen navigation = {navigation}/>)
      toJSONObject = toJSON
      logoutUser()
    })

    it('AuthLoadingScreen snapshot', () => {
      expect(toJSONObject()).toMatchSnapshot()
    })
  })
})
