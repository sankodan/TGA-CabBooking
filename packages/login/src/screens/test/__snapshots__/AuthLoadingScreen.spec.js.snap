// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLoadingScreen AuthLoadingScreen AuthLoadingScreen snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../packages/components/assets/background_dot.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "maxWidth": 340,
          "padding": 20,
          "width": "100%",
        },
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <ActivityIndicator
      color="#121330"
      size="large"
    />
  </View>
</View>
`;
